components/ntp_background_images/browser/view_counter_service.cc

```diff
@@ -174,7 +212,7 @@ void ViewCounterService::OnUpdated(NTPBackgroundImagesData* data) {
   // But keep view counter until branded content is seen.
   if (data) {
     model_.ResetCurrentWallpaperImageIndex();
```

Comment:
Ditto.
This callback is called when NTP SI/SR is called.
So, don't need to rest `current_wallpaper_image_index_` from the model.